frontends/common/parser_options.cpp

```diff
@@ -490,15 +490,18 @@ void ParserOptions::dumpPass(const char *manager, unsigned seq, const char *pass
             if (filename == "-") filename = "tmp.p4";
 
             cstring fileName = makeFileName(dumpFolder, filename, suffix);
-            auto stream = openFile(fileName, true);
+            std::unique_ptr<std::ostream> stream{openFile(fileName, true)};
             if (stream != nullptr) {
                 if (Log::verbose()) std::cerr << "Writing program to " << fileName << std::endl;
-                P4::ToP4 toP4(stream, Log::verbose(), file);
+                P4::ToP4 toP4(stream.get(), Log::verbose(), file);
                 if (noIncludes) {
                     toP4.setnoIncludesArg(true);
                 }
-                node->apply(toP4);
-                delete stream;  // close the file
+                if (node) {
+                    node->apply(toP4);
+                } else {
+                    *stream << "No P4 program returned by the pass" << std::endl;
```

Comment:
So this only happens when there is an error right? I would clarify this in the message.